### PR TITLE
(#84) Meta Data tests 

### LINKED ISSUE #84: Migration of Meta Data tests
## As a user, I want to make sure that correct metadata is used in the site.
> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin

Feature: MetaData Tests
    Scenario Outline: As an user, I want to see the correct OGType for metatags, DCTerms metatag content and general tests related to metatag, such as twitter tag, url, title etc...

        Given user is navigating to "<path>"
        Then page title is "<pageTitle>"
        And the page contains meta tags with the following properties
            | property       | content          |
            | og:title       | <og:title>       |
            | og:type        | <og:type>        |
            | og:url         | <og:url>         |
            | og:description | <og:description> |
            | og:site_name   | <og:site_name>   |

        And the meta description is "<metaDesc>"
        And the dcTermType is "<dcTermsType>"
        And the dcTermsSubject is "<dcTermsSubject>"
        And the dcTermsPartOf is "<dcTermsPartOf>"
        And the dcTermsCoverage is "<dcTermsCoverage>"
        And the dcTermsReferencedBy is "dcTermsReferencedBy>"
        And the twitter is "<twitter>"
        And the issuedDate is "<issuedDate>"

        Examples:
            | path                                                                                 | pageTitle                                                                                                                           | OGTitle                                                                                              | OGType  | og:url                                                                               | metaDesc                                                                                                                                                                                                                                                                                                               | ogDesc                                                                                                                                                                                                                                                                                                                 | dcTermsType          | dcTermsSubject            | dcTermsPartOf                          | og:siteName                   | dcTermsCoverage         | dcTermsReferencedBy | twitter | issuedDate                                                                       |
            | /about-cancer/coping/feelings/relaxation/stress-fact-sheet                           | Psychological Stress and Cancer                                                                                                     | Psychological Stress and Cancer - Browser Title                                                      | Website | /about-cancer/coping/feelings/relaxation/stress-fact-sheet                           | Psychological Stress and Cancer                                                                                                                                                                                                                                                                                        | Psychological Stress and Cancer                                                                                                                                                                                                                                                                                        | cgvArticle           | About Cancer              | Coping                                 | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/cancer/sobrellevar/self-image                                               | Su imagen propia y su sexualidad                                                                                                    | Self-Image, Sexuality, and Cancer                                                                    | Website | /espanol/cancer/sobrellevar/self-image                                               | Cómo hacer frente a la manera en que el cáncer y su tratamiento pueden cambiar la forma como usted se ve y se siente.                                                                                                                                                                                                  | Cómo hacer frente a la manera en que el cáncer y su tratamiento pueden cambiar la forma como usted se ve y se siente.                                                                                                                                                                                                  | cgvArticle           | About Cancer - Spanish    | Coping - Spanish                       | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /news-events/press-releases/2019/deep-learning-cervical-cancer-screening             | AI approach outperformed human experts in identifying cervical precancer                                                            | AI approach outperformed human experts in identifying cervical precancer                             | Website | /news-events/press-releases/2019/deep-learning-cervical-cancer-screening             | A research team led by investigators from the National Institutes of Health and Global Good has developed a computer algorithm that can analyze digital images of a woman’s cervix and accurately identify precancerous changes...                                                                                     | A research team led by investigators from the National Institutes of Health and Global Good has developed a computer algorithm that can analyze digital images of a woman’s cervix and accurately identify precancerous changes...                                                                                     | cgvPressRelease      | News and Events           | Press Releases                         | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/noticias/comunicados-de-prensa/2018/leucemia-llc-ibrutinib-estudio          | Leucemia - Ibrutinib más rituximab fue superior al tratamiento convencional para algunos pacientes con leucemia linfocítica crónica | Leucemia - Terapia basada en Ibrutinib fue superior al tratamiento convencional en un estudio de LLC | Website | /espanol/noticias/comunicados-de-prensa/2018/leucemia-llc-ibrutinib-estudio          | leukemia page description. - ES                                                                                                                                                                                                                                                                                        | leukemia page description. - ES                                                                                                                                                                                                                                                                                        | cgvPressRelease      | News and Events - Spanish | Press Releases - Spanish               | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /news-events/cancer-currents-blog/2019/vitamin-d-supplement-cancer-prevention        | Vitamin D Supplements Don’t Reduce Cancer Incidence, Trial Shows                                                                    | Vitamin D Supplements Don’t Reduce Cancer Incidence                                                  | Website | /news-events/cancer-currents-blog/2019/vitamin-d-supplement-cancer-prevention        | Blog Post page desc                                                                                                                                                                                                                                                                                                    | Blog Post page desc                                                                                                                                                                                                                                                                                                    | cgvBlogPost          | News and Events           | CancerCurrents                         | National Cancer Institute     | nciglobal,ncienterprise | event1,event53      | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/noticias/temas-y-relatos-blog/2019/vitamina-d-complemento-cancer-prevencion | Estudio indica que complementos de vitamina D no reducen la incidencia de cáncer                                                    | Complementos de vitamina D no reducen la incidencia de cáncer                                        | Website | /espanol/noticias/temas-y-relatos-blog/2019/vitamina-d-complemento-cancer-prevencion | Blog Post page desc                                                                                                                                                                                                                                                                                                    | Blog Post page desc                                                                                                                                                                                                                                                                                                    | cgvBlogPost          | News and Events - Spanish | CancerCurrentsEsp                      | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1,event53      | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /about-cancer/coping/feelings/relaxation/2019-investigators-site                     | 2019 Investigators' - Site Coordinators' Opportunity for Research Excellence (I-SCORE) Workshop                                     | 2019 Investigators' - Site Coordinators' Opportunity for Research Excellence (I-SCORE) Workshop      | Website | /about-cancer/coping/feelings/relaxation/2019-investigators-site                     | Information about the 2019 Investigators' - Site Coordinators' Opportunity for Research Excellence (I-SCORE) Workshop event.                                                                                                                                                                                           | Information about the 2019 Investigators' - Site Coordinators' Opportunity for Research Excellence (I-SCORE) Workshop event.                                                                                                                                                                                           | cgvEvent             | About Cancer              | Coping                                 | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /about-cancer/coping/feelings/relaxation/loukissas-jennifer                          | Jennifer K. Loukissas, M.P.P.                                                                                                       | Jennifer K. Loukissas, M.P.P., biosketch and professional background                                 | Website | /about-cancer/coping/feelings/relaxation/loukissas-jennifer                          | Jennifer K. Loukissas, M.P.P., biosketch and professional background                                                                                                                                                                                                                                                   | Jennifer K. Loukissas, M.P.P., biosketch and professional background                                                                                                                                                                                                                                                   | cgvBiography         | About Cancer              | Coping                                 | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /news-events/cancer-currents-blog                                                    | Cancer Currents Blog                                                                                                                | Cancer Currents Blog                                                                                 | Website | /news-events/cancer-currents-blog                                                    | cancer currents, blog                                                                                                                                                                                                                                                                                                  | cancer currents, blog                                                                                                                                                                                                                                                                                                  | cgvBlogSeries        | News and Events           | CancerCurrents                         | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /espanol/noticias/temas-y-relatos-blog                                               | Temas y relatos blog                                                                                                                | Temas y relatos blog                                                                                 | Website | /espanol/noticias/temas-y-relatos-blog                                               | cancer currents, blog                                                                                                                                                                                                                                                                                                  | cancer currents, blog                                                                                                                                                                                                                                                                                                  | cgvBlogSeries        | News and Events - Spanish | CancerCurrentsEsp                      | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /about-cancer/coping/feelings/relaxation/duke-university                             | Duke University Medical Center- Placeholder                                                                                         | DDuke University Medical Center- Placeholder                                                         | Website | /about-cancer/coping/feelings/relaxation/duke-university                             | Duke University Medical Center- Placeholder                                                                                                                                                                                                                                                                            | Duke University Medical Center- Placeholder                                                                                                                                                                                                                                                                            | cgvCancerCenter      | About Cancer              | Coping                                 | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /types/breast/research                                                               | Breast Cancer Research                                                                                                              | Breast Cancer Research                                                                               | Website | /types/breast/research                                                               | Find research articles on breast cancer, which may include news stories, clinical trials, blog posts, and descriptions of active studies.                                                                                                                                                                              | Find research articles on breast cancer, which may include news stories, clinical trials, blog posts, and descriptions of active studies.                                                                                                                                                                              | cgvCancerResearch    | Cancer Types              | Cancer Types Landing Page              | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/tipos/seno/investigacion                                                    | Investigación sobre el cáncer de seno (mama)                                                                                        | Investigación sobre el cáncer de seno (mama)                                                         | Website | /espanol/tipos/seno/investigacion                                                    | Spanish meta description                                                                                                                                                                                                                                                                                               | Spanish meta description                                                                                                                                                                                                                                                                                               | cgvCancerResearch    | Cancer Types - Spanish    | Cancer Types Landing Page - Spanish    | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /types/breast                                                                        | Breast Cancer—Patient Version                                                                                                       | Breast Cancer—Patient Version                                                                        | Website | /types/breast                                                                        | Breast cancer is the second most common cancer in women after skin cancer. Mammograms can detect breast cancer early, possibly before it has spread. Start here to find information on breast cancer treatment, causes and prevention, screening, research, and statistics.                                            | Breast cancer is the second most common cancer in women after skin cancer. Mammograms can detect breast cancer early, possibly before it has spread. Start here to find information on breast cancer treatment, causes and prevention, screening, research, and statistics.                                            | cgvCancerTypeHome    | Cancer Types              | Cancer Types Landing Page              | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /espanol/tipos/seno                                                                  | Cáncer de seno (mama)—Versión para pacientes                                                                                        | Cáncer de seno (mama)—Versión para pacientes                                                         | Website | /espanol/tipos/seno                                                                  | En las mujeres, el cáncer de mama es el segundo tipo de cáncer más común, seguido por el cáncer de piel. Las mamografías pueden detectar el cáncer de mama temprano. Comience aquí su búsqueda de información sobre tratamiento, causas y prevención, investigación y estadísticas relacionados con el cáncer de mama. | En las mujeres, el cáncer de mama es el segundo tipo de cáncer más común, seguido por el cáncer de piel. Las mamografías pueden detectar el cáncer de mama temprano. Comience aquí su búsqueda de información sobre tratamiento, causas y prevención, investigación y estadísticas relacionados con el cáncer de mama. | cgvCancerTypeHome    | Cancer Types - Spanish    | Cancer Types Landing Page - Spanish    | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | Null                                                                             |
            | /news-events                                                                         | News & Events                                                                                                                       | News & Events                                                                                        | Website | /news-events                                                                         | The latest cancer news from the U.S. government's principal agency for cancer research, plus resources designed for science writers and reporters.                                                                                                                                                                     | The latest cancer news from the U.S. government's principal agency for cancer research, plus resources designed for science writers and reporters.                                                                                                                                                                     | cgvHomeLanding       | News and Events           | News and Events Landing Page           | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/noticias                                                                    | Nuestro instituto                                                                                                                   | Noticias                                                                                             | Website | /espanol/noticias                                                                    | Las últimas noticias de la principal dependencia del gobierno de los Estados Unidos dedicada a la investigación del cáncer, así como recursos para reporteros y escritores científicos.                                                                                                                                | Las últimas noticias de la principal dependencia del gobierno de los Estados Unidos dedicada a la investigación del cáncer, así como recursos para reporteros y escritores científicos.                                                                                                                                | cgvHomeLanding       | News and Events - Spanish | News and Events Landing Page - Spanish | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /about-nci/organization/ccct                                                         | Coordinating Center for Clinical Trials (CCCT)                                                                                      | Coordinating Center for Clinical Trials (CCCT)                                                       | Website | /about-nci/organization/ccct                                                         | The Coordinating Center for Clinical Trials (CCCT) plays a key role in supporting NCI’s clinical and translational research programs to advance science and patient care.                                                                                                                                              | The Coordinating Center for Clinical Trials (CCCT) plays a key role in supporting NCI’s clinical and translational research programs to advance science and patient care.                                                                                                                                              | cgvMiniLanding       | About NCI                 | Organization                           | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /espanol/investigacion/iniciativas-clave                                             | Iniciativas clave                                                                                                                   | Key Initiatives                                                                                      | Website | /espanol/investigacion/iniciativas-clave                                             | Entérese acerca de las principales iniciativas de investigación creadas por el NCI para responder a cuestiones específicas de alta prioridad para la comunidad de investigación oncológica.                                                                                                                            | Entérese acerca de las principales iniciativas de investigación creadas por el NCI para responder a cuestiones específicas de alta prioridad para la comunidad de investigación oncológica.                                                                                                                            | cgvMiniLanding       | Research - Spanish        | Research Landing Page - Spanish        | Instituto Nacional del Cáncer | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|[12][0-9]\|3[01])[- /.](0[1-9]\|1[012])[- /.](20)\d\d - \d\d:\d\d$     |
            | /about-cancer/treatment/side-effects/skin-nail-changes/pruritus-pdq                  | Pruritus (PDQ®)–Patient Version                                                                                                     | Pruritus (PDQ®)–Patient Version                                                                      | Website | /about-cancer/treatment/side-effects/skin-nail-changes/pruritus-pdq                  | Pruritus (itching of the skin) can be caused by certain cancers or cancer treatment. Learn more about causes and treatment of pruritus in this expert-reviewed summary.                                                                                                                                                | Pruritus (itching of the skin) can be caused by certain cancers or cancer treatment. Learn more about causes and treatment of pruritus in this expert-reviewed summary.                                                                                                                                                | pdqCancerInfoSummary | About Cancer              | Treatment                              | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |
            | /about-cancer/treatment/drugs/blinatumomab                                           | Blinatumomab                                                                                                                        | Blinatumomab                                                                                         | Website | /about-cancer/treatment/drugs/blinatumomab                                           | This page contains brief information about blinatumomab and a collection of links to more information about the use of this drug, research results, and ongoing clinical trials.                                                                                                                                       | This page contains brief information about blinatumomab and a collection of links to more information about the use of this drug, research results, and ongoing clinical trials.                                                                                                                                       | pdqDrugInfoSummary   | About Cancer              | About Cancer (Other)                   | National Cancer Institute     | nciglobal,ncienterprise | event1              | summary | ^(0[1-9]\|1[012])[- /.](0[1-9]\|[12][0-9]\|3[01])[- /.](19\|20)\d\d - \d\d:\d\d$ |



```




### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*
* Requirements: https://github.com/NCIOCPL/cgov-digital-platform-qa/issues/90

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

## Solution

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### LINKED ISSUE #123: Migration of Search Restrictions tests for non indexing pages
## As a search engine, I want to check the search restrictions on pages with no index

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
    Scenario Outline: Check 'Exclude Search' is selected in pages with No Index
        Given user is navigating to "<path>"
        Then the page contains meta tags with the following names
            | robots                | noindex            |

        Examples:
            | path                                                                                        |
|/search/results?swKeyword=pact|
```
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

## Solution

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Added feature file.
Added tests.

Note: Noted issue #84  in #118 
Closes #84 
Closes #123

### CODE REVIEW

**Comment by alinai** [comment]
Test run results:
image

image

**Comment by alinai** [comment]
@KateMashkinaNIH , please review.
 FYI, in the old suite we had no column for og:url in the data provider. So added data for og:url and verified it.

File: cypress/integration/MetaData/MetaDataTests/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if (err.message.includes(`Cannot read property 'getAttribute' of undefined`)) {
+            return false
+        }
+    });
+    cy.visit(a);
+});
+
+And('the page contains og:site_name meta tags with the following properties', (dataTable) => {
+    for (const { property, content } of dataTable.hashes()) {
+        if (content.includes('{CANONICAL_HOST}')) {
```
Comment (KateMashkinaNIH) [inline cypress/integration/MetaData/MetaDataTests/index.js:19]: is og:site_name ever going to include cannonical host? this logic is for tags values that are built from path and host, 
I don't believe site_name is one of them

File: cypress/integration/MetaData/MetaDataTests/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            });
+        }
+    }
+});
+
+And('the page contains dcterms.issued meta tags with the following names', (dataTable) => {
+    cy.document().then((doc) => {
+        for (const { name, content } of dataTable.hashes()) {
+            if (content.includes('{CANONICAL_HOST}')) {
+                cy.location('host').then(host => {
```
Comment (KateMashkinaNIH) [inline cypress/integration/MetaData/MetaDataTests/index.js:40]: same here -  will it ever have a host ?

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/integration/MetaData/MetaDataTests/index.js (hunk 1 above)
Comment (alinai) [reply cypress/integration/MetaData/MetaDataTests/index.js]: Updated per review.

File: cypress/integration/MetaData/MetaDataTests/index.js (hunk 2 above)
Comment (alinai) [reply cypress/integration/MetaData/MetaDataTests/index.js]: Updated per review.

**Comment by alinai** [comment]
@KateMashkinaNIH , updated this one per your review. Thanks!

**Comment by alinai** [comment]
@KateMashkinaNIH , Added tests for #123 to this test per your suggestion. Please review. Thanks!

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**